vmdb/app/models/ext_management_system.rb

```diff
@@ -1,20 +1,26 @@
 class ExtManagementSystem < ActiveRecord::Base
+  SUBCLASSES = %w(
+    EmsInfra
+    EmsCloud
+    EmsContainer
+  )
+
   include EmsRefresh::Manager
 
   def self.types
-    [EmsInfra, EmsCloud].collect(&:types).flatten
+    leaf_subclasses.collect(&:ems_type)
   end
 
   def self.supported_types
-    [EmsInfra, EmsCloud].collect(&:supported_types).flatten
+    supported_subclasses.collect(&:ems_type)
```

Comment:
`supported_subclasses` and `leaf_subclasses` return an array, so normal collect.  In the flat_map case I am collecting Arrays,  so need to flatten.